pkg/stub/secretsChangeHandler_test.go

```diff
@@ -0,0 +1,173 @@
+	}
+	return &v1alpha1.MobileClientList{
+		Items: items,
+	}, nil
+}
+
+func (r *mockMobileClientRepo) DeleteByName(name string) error {
+	delete(r.mockStore, name)
+	return nil
+}
+
+func (r *mockMobileClientRepo) Watch() func() (watch.Interface, error) {
+	return nil
+}
+
+type mockWatchInterface struct{}
+
+func (m *mockWatchInterface) Stop() {}
+
+func (m *mockWatchInterface) ResultChan() <-chan watch.Event {
+	ch := make(chan watch.Event)
+	go func() {
+		ch <- watch.Event{
+			Object: nil,
+			Type:   "event",
+		}
+	}()
+	return ch
+}
+
+type mockSecretsCRUDL struct{}
+
+func NewMockSecretsCRUDL() *mockSecretsCRUDL {
+	return &mockSecretsCRUDL{}
+}
+
+func (m *mockSecretsCRUDL) List(namespace string) (*k8v1.SecretList, error) {
+	labels := make(map[string]string)
+	labels["mobile"] = "enabled"
+	labels["clientId"] = "test-app"
+	data := make(map[string][]byte)
+	data["id"] = []byte("test-service")
+	return &k8v1.SecretList{
+		Items: []k8v1.Secret{
+			k8v1.Secret{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:          labels,
+					ResourceVersion: "1",
+				},
+				Data: data,
+			},
+		},
+	}, nil
+}
+
+func (m *mockSecretsCRUDL) Create(namespace string, secret *k8v1.Secret) (*k8v1.Secret, error) {
+	return nil, nil
+}
+
+func (m *mockSecretsCRUDL) Watch(namespace string) func() (watch.Interface, error) {
+	return func() (watch.Interface, error) {
+		return &mockWatchInterface{}, nil
+	}
+}
+
+type mockSecretsCRUDL2 struct{}
+
+func NewMockSecretsCRUDL2() *mockSecretsCRUDL2 {
+	return &mockSecretsCRUDL2{}
+}
+
+func (m *mockSecretsCRUDL2) List(namespace string) (*k8v1.SecretList, error) {
+	labels := make(map[string]string)
+	labels["mobile"] = "enabled"
+	labels["clientId"] = "test-app"
+	data := make(map[string][]byte)
+	data["id"] = []byte("test-service")
+	return &k8v1.SecretList{
+		Items: []k8v1.Secret{},
+	}, nil
+}
+
+func (m *mockSecretsCRUDL2) Create(namespace string, secret *k8v1.Secret) (*k8v1.Secret, error) {
+	return nil, nil
+}
+
+func (m *mockSecretsCRUDL2) Watch(namespace string) func() (watch.Interface, error) {
+	return func() (watch.Interface, error) {
+		return &mockWatchInterface{}, nil
+	}
+}
+
+func TestHandler(t *testing.T) {
+	mockSecrets := NewMockSecretsCRUDL()
+	mockApps := NewMockMobileClientRepo()
+	client := &v1alpha1.MobileClient{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-app",
+		},
+		Spec: v1alpha1.MobileClientSpec{
+			Name: "test-app",
+		},
+		Status: v1alpha1.MobileClientStatus{},
+	}
+	mockApps.Create(client)
+	HandleSecretsChange("test-namespace", mockApps, mockSecrets)
+	app, _ := mockApps.ReadByName("test-app")
+	if len(app.Status.Services) != 1 {
+		t.Fatalf("mobile client validation failed")
+	}
+	mockSecrets2 := NewMockSecretsCRUDL2()
+	HandleSecretsChange("test-namespace", mockApps, mockSecrets2)
+	app, _ = mockApps.ReadByName("test-app")
+	if len(app.Status.Services) != 0 {
+		t.Fatalf("mobile client validation failed")
+	}
```

Comment:
if it's not too much trouble, can we use the same pattern for the test cases as we did in other tests. It will make it easier to see cases we have covered, and adding new test cases.